scripts/data-lint-lib.js

```diff
@@ -0,0 +1,187 @@
+t path = require("path");
+
+const Ajv = require("ajv");
+const addFormats = require("ajv-formats");
+const fastglob = require("fast-glob");
+
+const ajv = new Ajv({ allErrors: true });
+addFormats(ajv);
+
+module.exports = { schemaLint };
+
+async function schemaLint(dir = "", ext = "*.json", schema = {}) {
+	const validate = ajv.compile(schema);
+
+	const validFilesGlob = path.join(dir, ext);
+	// Find ALL files in the specified directory that do _NOT_ match the expected extension.
+	const invalidFiles = await fastglob([
+		path.join(dir, "*"),
+		`!${validFilesGlob}`
+	]);
+	const validFiles = await fastglob(validFilesGlob, {
+		absolute: true,
+		objectMode: true
+	});
+
+	const errors = [];
+
+	for (const invalidFile of invalidFiles) {
+		errors.push({
+			file: invalidFile,
+			errors: [
+				{
+					message: `Unexpected file type: "${path.extname(
+						invalidFile
+					)}". Expected "${ext}"`
+				}
+			]
+		});
+	}
+	for (const file of validFiles) {
+		const absPath = file.path;
+		const relPath = path.join(dir, file.name);
+		const contents = require(absPath);
+		await validate(contents).catch(err => {
+			const data = { file: relPath, errors: err.errors, contents };
+			errors.push(data);
+		});
+	}
+
+	if (errors.length) {
+		const err = new Error(`Schema check failed for ${validFilesGlob}`);
+		err.errors = errors;
+		return Promise.reject(err);
+	}
+	return;
+}
+
+/// SCHEMAS
```

Comment:
Not sure if we want to break this into yet another file (schemas.js). I can do that in a future refactor if we think its worth it, or if 3 files for a single task is excessive. Although it would let us potentially abstract all the AJV logic into a separate helper file.

But it'd make this file about 135 lines shorter. Heck, if we did `module.exports.communitySchema = ajv.compile({...})`, then this file wouldn't even need to know about ajv at all. It'd just have to take a directory name, expected file type glob, and a precompiled schema function. :shrug:

Reply:
I would break that down into a directory :-)

Reply:
I broke up the schemas into files now. Next iteration (not today) could be about the `ajv.compile()`.